samples/sensor/adt7420/sample.yaml

```diff
@@ -2,7 +2,9 @@ sample:
   name: ADT7420 Sensor Sample
 tests:
   sample.sensor.adt7420:
-    harness: sensor
+    pre-filter: dt_bus_enabled("i2c")
+    filter: dt_compat_enabled("adi,adt7420")
```

Comment:
Question on this:
If i2c is enabled in app overlay, the test config will pass the test pre-filter, right?

Reply:
As I understood it this worked based on pre-computed devicetree information from the board, which wouldn't be affected by shield or application overlays.  So it should probably be `board-dts-filter` or something like that, and it'd produce the wrong result if a board or application overlay added or removed a devicetree node or property that was part of the filter.

Reply:
So, from sanitycheck perspective, an application (or a shield) would not be able to enable matching device node, at least from sanitycheck point of view (while it's its responsibility to enable the matching Kconfig symbol). 
This would differ from the "manual/west" compilation way, which might be confusing. So it would have to be made clear for sample developers (and reviewers).

Reply:
> As I understood it this worked based on pre-computed devicetree information from the board, which wouldn't be affected by shield or application overlays. So it should probably be `board-dts-filter` or something like that,

correct.

> and it'd produce the wrong result if a board or application overlay added or removed a devicetree node or property that was part of the filter.

This is primarily to avoid building something that we know is not going to be supported, if a device does not have GPIO, then we can't build an application that tries to enable GPIO, this is what depends_on was trying to do, but not effectively. If a shield adds some GPIO, then this is really the corner case that most likely will be less than a fraction of cases we have.

We should choose our battles IMO and this is all to increase test coverage that we do not have right now. When all of this is done and complete we will have 1000s of additional builds and runs on devices that we never had before because of the static depends_on filtering.